src/librustc_codegen_llvm/builder.rs

```diff
@@ -494,9 +527,80 @@ impl Builder<'a, 'll, 'tcx> {
         }
     }
 
+    fn load_ref(
```

Comment:
This should be named `load_operand`. I guess the metadata is gnarly enough to merit moving this here. But maybe this should be in a separate `Methods` trait?